enricher/epss/epss.go

```diff
@@ -0,0 +1,391 @@
+	}
+
+	etag := resp.Header.Get("etag")
+	if etag == "" {
+		return nil, "", fmt.Errorf("ETag not found in response headers")
+	}
+
+	newFingerprint := driver.Fingerprint(etag)
+
+	if prevFingerprint == newFingerprint {
+		zlog.Info(ctx).Str("fingerprint", string(newFingerprint)).Msg("file unchanged; skipping processing")
+		return nil, prevFingerprint, nil
+	}
+
+	gzipReader, err := gzip.NewReader(resp.Body)
+	if err != nil {
+		return nil, "", fmt.Errorf("unable to decompress file: %w", err)
+	}
+	defer gzipReader.Close()
+
+	csvReader := csv.NewReader(gzipReader)
+	csvReader.FieldsPerRecord = 2
+
+	// assume metadata is always in the first line
+	record, err := csvReader.Read()
+	if err != nil {
+		return nil, "", fmt.Errorf("unable to read metadata line: %w", err)
+	}
+
+	var modelVersion, date string
+	for _, field := range record {
+		field = strings.TrimSpace(field)
+		field = strings.TrimPrefix(strings.TrimSpace(field), "#")
+		key, value, found := strings.Cut(field, ":")
+		if !found {
+			return nil, "", fmt.Errorf("unexpected metadata field format: %q", field)
+		}
+		switch key {
+		case "model_version":
+			modelVersion = value
+		case "score_date":
+			date = value
+		}
+	}
+
+	if modelVersion == "" || date == "" {
+		return nil, "", fmt.Errorf("missing metadata fields in record: %v", record)
+	}
+	csvReader.Comment = '#'
+	csvReader.FieldsPerRecord = 3 // Expect exactly 3 fields per record
+
+	if modelVersion == "" || date == "" {
+		return nil, "", fmt.Errorf("missing metadata fields in record: %v", record)
+	}
+
+	// Read and validate header line
+	record, err = csvReader.Read()
+	if err != nil {
+		return nil, "", fmt.Errorf("unable to read header line: %w", err)
+	}
+
+	expectedHeaders := []string{"cve", "epss", "percentile"}
+	if !slices.Equal(record, expectedHeaders) {
+		return nil, "", fmt.Errorf("unexpected CSV headers: %v", record)
+	}
+
+	enc := json.NewEncoder(out)
+	totalCVEs := 0
+
+	for {
+		record, err := csvReader.Read()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return nil, "", fmt.Errorf("unable to read line in CSV: %w", err)
+		}
+
+		r, err := newItemFeed(record, modelVersion, date)
+		if err != nil {
+			zlog.Warn(ctx).Err(err).Msg("skipping invalid record")
+			continue
+		}
+
+		if err := enc.Encode(&r); err != nil {
+			return nil, "", fmt.Errorf("unable to write JSON line to file: %w", err)
+		}
+		totalCVEs++
+	}
+
+	zlog.Info(ctx).Int("totalCVEs", totalCVEs).Msg("processed CVEs")
+	if _, err := out.Seek(0, io.SeekStart); err != nil {
+		return nil, newFingerprint, fmt.Errorf("unable to reset file pointer: %w", err)
+	}
+	success = true
+
+	return out, newFingerprint, nil
+}
+
+// ParseEnrichment implements driver.EnrichmentUpdater.
+func (e *Enricher) ParseEnrichment(ctx context.Context, rc io.ReadCloser) ([]driver.EnrichmentRecord, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "enricher/epss/Enricher/ParseEnrichment")
+
+	defer func() {
+		_ = rc.Close()
+	}()
```

Comment:
I don't think this is needed. From what I can tell, the caller should already `Close` the `io.ReadCloser`. I do see this is called in the CVSS enricher, though... Perhaps it's ok, though maybe just do `defer rc.Close()`

Unrelated, but it's a bit odd this takes in an `io.ReadCloser`. I think it can just be an `io.Reader`, but that's outside the scope of this PR